contracts/WethLoanRouter.sol

```diff
@@ -0,0 +1,88 @@
+pragma solidity ^0.8.3;
+
+import "./interfaces/IWethLoanRouter.sol";
+import "./interfaces/ILoanRouter.sol";
+import "./interfaces/IBondController.sol";
+import "./interfaces/ITranche.sol";
+import "./interfaces/IButtonWrapper.sol";
+// ToDo: @mark-toda, Do you want me to import this from button-wrappers or copy interface here
+import "./interfaces/IWETH.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @dev Weth Loan Router built on top of a LoanRouter of your choosing
+ * to allow loans to be created with raw ETH instead of WETH
+ */
+contract WethLoanRouter is IWethLoanRouter {
+    ILoanRouter public loanRouter;
```

Comment:
add `immutable` keyword - pretty big gas savings as it allows the compiler to essentially embed these values into the code rather than read from storage every time